tpm2/test/tpm2_test.go

```diff
@@ -1656,3 +1656,117 @@ func TestPolicyPCR(t *testing.T) {
 		})
 	}
 }
+
+func TestDictionaryAttackParameters(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	testCases := []struct {
+		name            string
+		maxTries        uint32
+		recoveryTime    uint32
+		lockoutRecovery uint32
+	}{
+		{
+			name:            "0 0 0",
+			maxTries:        0,
+			recoveryTime:    0,
+			lockoutRecovery: 0,
+		},
+		{
+			name:            "5 10 10",
+			maxTries:        5,
+			recoveryTime:    10,
+			lockoutRecovery: 10,
+		},
+	}
+
+	auth := AuthCommand{
+		Session:    HandlePasswordSession,
+		Attributes: AttrContinueSession,
+		Auth:       []byte(emptyPassword),
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if err := DictionaryAttackParameters(rw, auth, tc.maxTries, tc.recoveryTime, tc.lockoutRecovery); err != nil {
+				t.Fatalf("DictionaryAttackParameters failed: %v", err)
+			}
+			caps, _, err := GetCapability(rw, CapabilityTPMProperties, 3, uint32(MaxAuthFail))
+			if err != nil {
+				t.Fatalf("GetCapability failed: %v", err)
+			}
+			if tc.maxTries != caps[0].(TaggedProperty).Value {
+				t.Fatalf("expected %d, but got %d", tc.maxTries, caps[0].(TaggedProperty).Value)
```

Comment:
Ditto [got-before-want](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures) failure message order plz, here and below